### PR TITLE
[machinelearning] update llama.cpp user guide

### PR DESCRIPTION
**Pull request by snadampal**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: machinelearning/llama.cpp.md [hunk 1]
```diff
@@ -27,46 +27,79 @@ aws ec2 describe-images --owners amazon --filters "Name=architecture,Values=arm6
...
+The following instructions use Meta Llama-3 8B parameter model from [Hugging Face](https://huggingface.co/models) models repository to demonstrate LLM inference performance on AWS Graviton based EC2 Instances. The model is already availble in multiple quantized formats which can be directly run on AWS Graviton processors.
```
Comment (RamaMalladiAWS) [inline machinelearning/llama.cpp.md:50]: Can you add URL for the model? Is it this one: https://huggingface.co/meta-llama/Meta-Llama-3-8B? Thanks

File: machinelearning/llama.cpp.md [hunk 2]
```diff
@@ -27,46 +27,79 @@ aws ec2 describe-images --owners amazon --filters "Name=architecture,Values=arm6
...
 
 ```
 
 # Additional Resources
 
 Please refer to
 1. [Best-in-class LLM performance on Arm Neoverse V1 based AWS Graviton3 CPUs](https://community.arm.com/arm-community-blogs/b/infrastructure-solutions-blog/posts/best-in-class-llm-performance) to know the LLM inference performance measured on AWS Graviton3 based EC2 Instances.
-2. [Run LLMs on CPU with Amazon SageMaker Real-time Inference](https://community.aws/content/2eazHYzSfcY9flCGKsuGjpwqq1B/run-llms-on-cpu-with-amazon-sagemaker-real-time-inference?lang=en) for running LLMs for real-time inference using AWS Graviton3 and Amazon SageMaker.
+2. [Running Llama 3 70B on the AWS Graviton4 CPU with Human Readable Performance](https://community.arm.com/arm-community-blogs/b/infrastructure-solutions-blog/posts/running-llama-3-70b-on-aws-graviton4) for LLM inference performance on AWS Graviton4 based EC2 Instances.
+3. [Intro to Llama on Graviton](https://dev.to/aws-heroes/intro-to-llama-on-graviton-1dc) for a step by step guide on how to deploy an LLM model on AWS Graviton-based EC2 Instances.
```
Comment (RamaMalladiAWS) [inline machinelearning/llama.cpp.md:104]: This page has quantization that is specific to each version of Graviton. Would you still want to leave this reference in here? Maybe add a note that quantization isn't applicable with the latest `llama` release.
```
./llama-quantize --allow-requantize dolphin-2.9.4-llama3.1-8b-Q4_0.gguf dolphin-2.9.4-llama3.1-8b-Q4_0_8_8.gguf Q4_0_8_8
```

**Review by RamaMalladiAWS — APPROVED** [review]

**Review by ddynwzh1992 — APPROVED** [review]
Looks good to me

**Review by ddynwzh1992 — APPROVED** [review]

File: machinelearning/llama.cpp.md (hunk 2 above)
Comment (snadampal) [reply machinelearning/llama.cpp.md:104]: good point, let me add.

**Comment by snadampal** [comment]
addressed the comments, and merging it now.

**Pull request merged**